nestkernel/event_delivery_manager.h

```diff
@@ -52,6 +52,98 @@ typedef MPIManager::OffGridSpike OffGridSpike;
 class TargetData;
 class SendBufferPosition;
 
+/*
+ * A thread-safe array to coordinate progress across threads during
+ * gather operations.
+ */
+class CompletedChecker
+{
+private:
+  bool* a_;
```

Comment:
Why don't you use `vector<bool>` here? As far as I know, the C++ standard prescribes an efficient implementation for that specialization of `vector`.

Reply:
yes, you are right, but I haven't found a way to set the element of a vector after a `omp atomic` statement, see [here](https://www.ibm.com/support/knowledgecenter/SSGH2K_13.1.2/com.ibm.xlc1312.aix.doc/compiler_ref/prag_omp_atomic.html) for the supported operations. The other option would be to use a vector and `omp critical`, but I chose the former since I expect it to have better performance. however, I have not data to back this up.

Reply:
Thank you for the explanation! I think it would be useful to document the design decision in the code.